### DISCUSSION TITLE
Reference

### DISCUSSION THREAD

**Discussion by alexandster** [BenWiseman/sentiment.ai#5] [General]
Dear Ben
Newbie here. Thanks for putting out this excellent work. Could you point me towards resources that offer details on the methodology, i.e. regarding the Universal Sentence Encoder models ("en.large"). I would like to know how the sentiment scores are calculated by the default secondary scoring model from xgboost. Where can I read up on it? I can make your code run, but I don't completely understand what it does :-)

Cheers!
Alexandster

**Comment by BenWiseman** [comment]
Sure, so the TL;DR is that I pre-trained and xgb and glm on some labelled data. The data can't be open-sourced as it has some stuff from the funding agency/my employer, but basically, by converting text to an embedding vector where each element represents some quality of language, you can then use standard modelling techniques treating each element of the embedding vector as an independent variable. So we included some pre-trained models, and hopefully over time they will get even better!

**Reply by alexandster** [reply]
Sounds good, very clear! I understand the data restrictions, but can you describe the labelled data in terms of its quantity (X documents) and/or source (tweets, Wikipedia, product reviews) without making your employer angry? I am trying to assess whether I can apply sentiment.ai in a scientific study, so any information that helps readers understand the process would help.

**Reply by BenWiseman** [reply]
We use a variety of sources including established sentiment word  dictionaries (to mimic existing algos), employee feedback, IMDB reviews, product reviews, tweets, snippets of news articles, synthetic examples, etc. We had about 40,000 labelled examples in all, and tried to balance out different sources. Will do a more formal write up, but as for whether or not you can use it in a study, it all comes down to whether or not it validates well on your specific data -- which is something you'll need to assess no matter what package or technique you use!
